video/external/src/decoder/openh264.rs

```diff
@@ -1,13 +1,62 @@
 use core::slice;
 use std::ffi::{c_int, c_uchar};
+use std::fmt::Display;
 use std::ptr;
+use std::sync::Arc;
 
 use crate::decoder::openh264_sys::{self, videoFormatI420, ISVCDecoder, OpenH264};
 use crate::decoder::VideoDecoder;
 
 use ruffle_render::bitmap::BitmapFormat;
 use ruffle_video::error::Error;
 use ruffle_video::frame::{DecodedFrame, EncodedFrame, FrameDependency};
+use thiserror::Error;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct H264Version(u32, u32, u32);
+
+impl Display for H264Version {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "({},{},{})", self.0, self.1, self.2)?;
+        Ok(())
+    }
+}
+
+#[derive(Debug, Error)]
+pub enum H264Error {
+    #[error("Error while loading OpenH264: {0}")]
+    LibraryLoadingError(#[from] ::libloading::Error),
+
+    #[error("OpenH264 version mismatch, expected {0}, was {1}")]
+    VersionMismatchError(H264Version, H264Version),
+}
+
+/// H264 codec representation.
+pub struct H264Codec {
+    openh264: Arc<OpenH264>,
+}
+
+impl H264Codec {
+    const VERSION: H264Version = H264Version(2, 4, 1);
+
+    pub fn new<P>(filename: P) -> Result<Self, H264Error>
+    where
+        P: AsRef<::std::ffi::OsStr>,
```

Comment:
Again, feels strange to call the type of `filename` `P`, of all letters, when it's not even a `PathBuf`.

Reply:
Resolving for the same reason as above, consistency with `libloading`.